tree_queries/compiler.py

```diff
@@ -139,6 +139,12 @@ def as_sql(self, *args, **kwargs):
         }
 
         if "__tree" not in self.query.extra_tables:  # pragma: no branch - unlikely
+            tree_params = params.copy()
```

Comment:
Is this necessary? The `params` dictionary is our object and is only used right here in these lines.

If I read your pull request correctly we could always use `base_table` if it is set and use `db_table` as a fallback, something like `params = { .... , "db_table": self.query.base_table or opts.db_table, ..... }` 

What do you think?

Reply:
It is because we are using the params to build the actual SQL, and by changing the table name to the aliased one used in subqueries, the SQL fail. (sorry my database terms are not strong enough..)

So, instead of this sub query:
```sql
    WITH RECURSIVE __tree(tree_depth, tree_path, tree_ordering, tree_pk) AS (
        SELECT
            0 tree_depth,
            printf("%s", id) tree_path,
            printf("%020s", position) tree_ordering,
            T."id" tree_pk
        FROM testapp_model T
        WHERE T."parent_id" IS NULL

        UNION ALL

        SELECT
            __tree.tree_depth + 1,
            __tree.tree_path || printf("%s", T.id),
            __tree.tree_ordering || printf("%020s", T.position),
            T."id"
        FROM testapp_model T
        JOIN __tree ON T."parent_id" = __tree.tree_pk
    )
```
we'll get this:
```sql
    WITH RECURSIVE __tree(tree_depth, tree_path, tree_ordering, tree_pk) AS (
        SELECT
            0 tree_depth,
            printf("%s", id) tree_path,
            printf("%020s", position) tree_ordering,
            T."id" tree_pk
        FROM U0 T # notice here
        WHERE T."parent_id" IS NULL

        UNION ALL

        SELECT
            __tree.tree_depth + 1,
            __tree.tree_path || printf("%s", T.id),
            __tree.tree_ordering || printf("%020s", T.position),
            T."id"
        FROM U0 T # notice here
        JOIN __tree ON T."parent_id" = __tree.tree_pk
    )
```

Reply:
Ah now I get it. Thanks for the explanation. I interpreted UX as User eXperience or something and got confused.